.changeset/good-spies-share.md

```diff
@@ -0,0 +1,46 @@
+---
+'@apollo/server': minor
+---
+
+# Added ability to use a custom executor such as graphql-jit
+
+Apollo Server uses graphql-js behind the scenes when executing graphql queries.
+If you would like to use a custom executor, such as
+[graphql-jit](https://github.com/zalando-incubator/graphql-jit/tree/main),
+you can swap in a `customExecutor`. This can result in significant performance
+gains when used in combination with `staticOnly` cache control hints, especially with larger response sizes.
+
+Note that this cannot be used when using the `gateway` option.
+
+```ts
+import { compileQuery, isCompiledQuery } from 'graphql-jit';
+import { lru } from 'tiny-lru';
+
+const executor = (schema: GraphQLSchema, cacheSize = 2014, compilerOpts = {}) => {
```

Comment:
There's a schema on GraphQLRequestContext too — can we just use that one instead of having to specify it both to this function and to the server? I guess perhaps we then have to include it in the cache key?

Reply:
Are you suggesting using schema from `request` instead of passing it into the executor?

I suppose that could work assuming its the same schema object.